### PR TITLE
[#751] add support for rewind_to_height to FsBlockDb

### LINKED ISSUE #751: add support for rewind_to_height to `FsBlockDb`
a BlockSource might be rewound in different scenarios like chain re-orgs or user triggered re-scans.

Whatever the case, wallets need BlockSource implementations to be able to go back a few blocks to resume scanning. 

Reorgs would mean a max rewind distance of 100 blocks, whereas quick re-scans will depend on the wallet's state. On current linear ascending scan, quick re-scans go back to the minedHeight of the oldest unspent note.

the rust sqlite_crate assumes that caller will delete the files rewound and will roll back the database only.

### PR DESCRIPTION
**Pull request by pacu**
Closes #751

### CODE REVIEW

File: zcash_client_sqlite/src/chain.rs [hunk 1]
```diff
@@ -139,6 +139,17 @@ pub(crate) fn blockmetadb_insert(
     }
 }
 
+#[cfg(feature = "unstable")]
+pub(crate) fn blockmetadb_rewind_to_height(
+    conn: &Connection,
+    block_height: BlockHeight,
+) -> Result<(), rusqlite::Error> {
+    conn.prepare("DELETE FROM compactblocks_meta WHERE height > ?")
+        .unwrap()
```
Comment (nuttycom) [inline zcash_client_sqlite/src/chain.rs:148]:
```suggestion
    conn.prepare("DELETE FROM compactblocks_meta WHERE height > ?")?
```

**Review by nuttycom — CHANGES_REQUESTED** [review]
This looks fine apart from the `unwrap`.

File: zcash_client_sqlite/src/chain.rs [hunk 2]
```diff
@@ -139,6 +139,16 @@ pub(crate) fn blockmetadb_insert(
     }
 }
 
+#[cfg(feature = "unstable")]
+pub(crate) fn blockmetadb_rewind_to_height(
+    conn: &Connection,
+    block_height: BlockHeight,
+) -> Result<(), rusqlite::Error> {
+    conn.prepare("DELETE FROM compactblocks_meta WHERE height > ?")
```
Comment (nuttycom) [inline zcash_client_sqlite/src/chain.rs:147]:
```suggestion
    conn.prepare("DELETE FROM compactblocks_meta WHERE height > ?")?
```

Comment (nuttycom) [reply zcash_client_sqlite/src/chain.rs:147]: Missed the question mark (to return the result as an error) rather than `unwrap`.

**Review by nuttycom — CHANGES_REQUESTED** [review]
Not compiling.

**Comment by str4d** [comment]
Compilation failure is due to `rustls 0.20.8` bumping its MSRV to 1.57.0 (as a result of its own dependencies bumping MSRV). We only use `rustls` when the `download-params` feature of `zcash_proofs` is enabled; here it is being enabled during testing.

I'll open a separate issue for handling or bumping MSRV.

**Review by nuttycom — APPROVED** [review]
utACK

**Pull request merged**